Wikipedia/Code/Session.swift

```diff
@@ -210,11 +212,18 @@ import Foundation
     }
     
     public func dataTask(with request: URLRequest, callback: Callback) -> URLSessionTask {
+        
+        //tonitodo: may try a wrapper method here of some sort for SchemeHandler, if request header forceCache is true, return from cache and avoid urlsession task resume altogether. else kickoff the session task.
```

Comment:
I might be missing something, but could the presence or lack of presence of the cache key header be an indicator of whether or not to check the persistent cache instead of a separate `forceCache` header?

Reply:
@joewalsh yeah, this was for handling that case where they restore from NavigationStateController on launch - don't even try to fetch anything first so they see the article as quickly as possible. Standard handling if the cache key exists in the header would be to still try to fetch then fallback to cache if there's an error or a 304, so they'd still see a loading bar. I think rather than a forceCache flag better would be assign NSURLRequest.CachePolicy here to .returnCacheDataElseLoad and check for that here, falling back to PersistentCache if URLCache doesn't exist.

Reply:
Sounds good - I forgot about that case (again)!